src/app/ReadClient.cpp

```diff
@@ -894,32 +904,46 @@ void ReadClient::OnLivenessTimeoutCallback(System::Layer * apSystemLayer, void *
                  "Subscription Liveness timeout with SubscriptionID = 0x%08" PRIx32 ", Peer = %02x:" ChipLogFormatX64,
                  _this->mSubscriptionId, _this->GetFabricIndex(), ChipLogValueX64(_this->GetPeerNodeId()));
 
+    if (_this->mIsPeerICD)
+    {
+        // If the device is idle, we mark the subscription as "IdleSubscription", and trigger resubscription on
+        // `OnActiveModeNotification`. Note: the liveness timeout is always longer than the MaxInterval (and idle duration), so we
+        // can move the device to `IdleSubcription` when liveness timeout is reached.
+        ChipLogProgress(DataManagement, "Peer is not active now, mark the subscription as IdleSubscription.");
+        _this->MoveToState(ClientState::IdleSubscription);
+        return;
```

Comment:
Here's the thing: this is not notifying the ReadClient consumer that the subscription has dropped (calling OnResubscriptionNeeded, etc).  That seems odd; it violates existing client expectations.  It would _definitely_ break the Apple clients I am aware of.

Maybe the right thing is to do the Close() call as we normally do, etc, let the client decide whether they even _want_ to resubscribe (something this PR is not doing right either) but either have ScheduleResubscription not schedule things or have the things it schedules be a no-op?

In either case, existing notificaitons need to continue to work and existing logic that calls TriggerResubscribeIfScheduled should work to trigger an immediate resubscribe attempt: that logic is called when we know the device is awake (even if it has not sent a check-in yet), and needs to keep working....

Reply:
yes, agree, we still need notify the ReadClient consumer that the subscription has dropped.

Reply:
updated with two changes
-- notify the ReadClient consumer that the subscription has dropped with new Error code, CHIP_ERROR_ICD_SUBSCRIBE_INACTIVE_TIMEOUT, and let application decide which resubscription strategy they wanna go with.
-- Update DefaultResubscribePolicy not to go with resubscribe when the error is CHIP_ERROR_ICD_SUBSCRIBE_INACTIVE_TIMEOUT